js/reference.js

```diff
@@ -544,6 +564,62 @@ var ERMrest = (function(module) {
                     }
                     this._location.sortObject = sortObject;
                 }
+ 
+                /*
+                * Change api to attributegroup for retrieving the foreign key data
+                * This will just affect the http request and not this._location
+                *
+                * NOTE: 
+                * This piece of code is dependent on the same assumptions as the current parser, which are:
+                *   1. There is no alias in url (more precisely `M`, `F1`, `F2`, `F3`, ...)
+                *   2. Filter comes before the link syntax.
+                *   3. There is no trailing `/` in uri (as it will break the ermrest too).
+                */
+                if (this._table.foreignKeys.length() > 0) {
+                    var compactPath = this._location.compactPath,
+                        parts = compactPath.split('/'),
+                        tableIndex = 0,
+                        fkList = "",
+                        linking,
+                        sortColumn,
+                        keys,
+                        k;
+
+                    // add M alias to current table
+                    linking = parts[parts.length-1].match(/(\(.*\)=\(.*:.*:.*\))/);
+                    if (linking && linking[1]) { // the same logic as parser for finding the link syntax
```

Comment:
Just note to myself and @RFSH: Update this line to take search filter into account. See latest parser code.